### PR TITLE
Added initial JSDocs for pipeline stages

### PR DESCRIPTION
**Pull request by JoshuaStorm**
As a part of #130 

Still need to add docs for 'intermediate' steps of the stages. 

Unsure if I should be using hyphens following the parameter names, they're in the JSDocs documentation, but not in the `quantizeAttributes` that was already documented.

### CODE REVIEW

File: lib/addDefaults.js [hunk 1]
```diff
@@ -10,6 +10,53 @@ var findAccessorMinMax = require('./findAccessorMinMax');
 
 module.exports = addDefaults;
 
+/**
+ * Adds default values to any missing glTF values of the hierarchy.
+ *
+ * @param {Object} gltf - An object holding a glTF hierarchy
```
Comment (lilleyse) [inline lib/addDefaults.js]: Even though Cesium doesn't use the hyphens I kind of like them here. @pjcozzi do you have a preference?

File: lib/combineMeshes.js [hunk 2]
```diff
@@ -3,8 +3,16 @@ var Cesium = require('cesium');
 var defined = Cesium.defined;
 module.exports = combineMeshes;
 
-//Combines meshes in a node. 
-//Should be run before the combinePrimitives stage to merge all primitives for a node into one mesh.
+
+
```
Comment (lilleyse) [inline lib/combineMeshes.js]: Remove extra space.

File: lib/encodeImages.js [hunk 3]
```diff
@@ -4,6 +4,12 @@ var Jimp = require('jimp');
 
 module.exports = encodeImages;
 
+/**
+ * Encode images to a glTF buffer
+ *
+ * @param {Object} gltf - An object holding a glTF hierarchy
+ * @param {Function} callback
```
Comment (lilleyse) [inline lib/encodeImages.js]: The believe callbacks are documented like this: http://usejsdoc.org/tags-callback.html. At the same time, we will eventually change the callbacks to promises anyways.

File: lib/removeUnused.js [hunk 4]
```diff
@@ -16,8 +16,16 @@ var removeUnusedShaders = require('./removeUnusedShaders');
 
 module.exports = removeUnused;
 
+/**
+ * Removes unused values in a a glTF hierarchy.
+ * 
+ * @param {Object} gltf - An object holding a glTF hierarchy
+ * @param {Object} stats - An OptimizationStatistics Object
```
Comment (lilleyse) [inline lib/removeUnused.js]: Instead of {Object} make it {OptimizationStatistics}

File: lib/removeUnused.js [hunk 5]
```diff
@@ -16,8 +16,16 @@ var removeUnusedShaders = require('./removeUnusedShaders');
...
+ * 
+ * @returns glTF without redundant objects
```
Comment (lilleyse) [inline lib/removeUnused.js]: This description is a bit confusing.

File: lib/removeUnusedVertices.js [hunk 6]
```diff
@@ -12,7 +12,13 @@ var writeBufferComponentType = require('./writeBufferComponentType');
 
 module.exports = removeUnusedVertices;
 
-//Removes attributes from indexed primitives that aren't used
+/**
+ * Removes attributes from indexed primitives that aren't used
+ *
+ * @param {Object} gltf - An object holding a glTF hierarchy
+ * 
+ * @returns glTF without unnecessary vertices
```
Comment (lilleyse) [inline lib/removeUnusedVertices.js]: Change unnecessary to unused just to match the stage name.

File: lib/addDefaults.js (hunk 1 above)
Comment (lilleyse) [inline lib/addDefaults.js]: For the sake of consistency with other areas in this project, let's just remove the hyphens.

**Comment by JoshuaStorm** [comment]
I'm going to hold off on this until we have the majority of these converted to promise patterns

**Comment by lilleyse** [comment]
Closed in favor of #157

**Pull request closed without merging**